test-expiration

```diff
@@ -5,11 +5,30 @@ set -euxo pipefail
 TEST_CONFIG=${TEST_CONFIG:-./test-config.sh}
 . "$TEST_CONFIG"
 
+# The --show-keys option is only available on recent GPG2. Fallback to
+# --list-keys with --keyring and various options otherwise.
+if ${GPG} --dump-options | grep -q -x -e --show-keys; then
```

Comment:
On the one hand, I'm sure this works, and it's nice to know what the modern syntax is. On the other hand, if the rather verbose fallback syntax works on gpg2, could we just use that everywhere?

Reply:
Yeah, I suppose you're right. I was really happy when I found out about `--show-keys` because I spend so much time with gpg being like JUST SHOW ME WHAT I WANT. I'll change it to use the old syntax with a `FIXME` about what version is required to just use `--show-keys` instead.

Reply:
Per https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=NEWS;hb=HEAD, it looks like `--show-keys` was added in 2.2.8 but 2.2.9 had some bugfixes for it including implicitly enabling the `show-*` options.